### DISCUSSION TITLE
Bookmark Function

### DISCUSSION THREAD

**Discussion by mohbusaidy** [neuefische/questions#34] [Web] [answered]
Moinsen, 

Bei mir funktioniert meine bookmark function so halb 😅 es wird geaddet zu meinen favorites fled, aber eigentlich hatte ich erwartet das sich meine karte in das favorite feld hinzufügt, leider klappt das nicht. Es taucht ausschlieslich der name welche das alt="name" auf. Ich hab mich schon seit über das WE daran aufgehangen. 

Hier der link zu meinem branch: [favorite branch](https://github.com/mohbusaidy/Capstone-Project/tree/favorites)

**Comment by ahohnsen** [comment, marked as answer]
Hallo Mohammed,

Wenn ich deine Frage richtig verstanden habe, geht es darum, wie deine Favoriten angezeigt werden. Momentan steht in deiner `Favorites.js` folgendes:
```jsx
 {favoriteItems.map(item => (
        <div key={item.id}>
          <div>{item.name}</div>
        </div>
      ))}
```

Hier wird für deine Favoriten lediglich der Name ausgegeben mit `item.name`.
Wenn du stattdessen aber für jeden Favoriten eine Karte anzeigen lassen möchtest, genau wie die Karten oberhalb deiner Favoriten, dann kannst du folgendes tun:

Importiere deine Product-Komponente in deine `Favorites.js` und render diese in deiner Map-Funktion für jeden einzelnen Favoriten, genau wie du es bereits in der `Main.js` implementiert hast.

Ich hoffe, das hilft dir weiter 🙂

**Reply by mohbusaidy** [reply]
Danke! ich häng aber immer noch fest, weil ich jetzt ja meine Product Komponente importiere aber nirgends im code aufrufe, wenn ich das wie in der main koponente machen soll. In der Main komponente hatte ich ja "Product" mit styled.component aufgerufen sodass er meine Product komponente aufgerufen wurde. 

<img width="300px" alt="Bildschirmfoto 2022-08-01 um 15 30 25" src="https://user-images.githubusercontent.com/104504053/182160161-4178de06-e55b-4008-b900-4cfa0ee67f42.png">

**Reply by ahohnsen** [reply]
In deiner `Main.js` rufst du deine Product-Komponente in deiner Map-Funktion auf:

```jsx
{products.map(product => (
          <Product key={product.id} product={product} onAdd={onAdd} />
        ))}
```

Genau so kannst du auch in deiner `Favorites.js` verfahren:

```jsx
{favoriteItems.map(item => (
        <Product key={item.id} product={item} onAdd={onAdd} />
      ))}
```
